components/lib/crash/src/main/java/mozilla/components/lib/crash/notification/CrashNotification.kt

```diff
@@ -28,42 +31,36 @@ internal class CrashNotification(
 ) {
     fun show() {
         val pendingIntent = PendingIntent.getActivity(
-            context, 0, CrashPrompt.createIntent(context, crash), 0
+                context, PROMPT_REQUEST_CODE, CrashPrompt.createIntent(context, crash), 0
         )
 
-        val channel = ensureChannelExists()
+        val reportPendingIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            PendingIntent.getForegroundService(
+                    context, REPORT_REQUEST_CODE, CrashPrompt.createReportIntent(context, crash), 0
+            )
+        } else {
+            PendingIntent.getService(
+                    context, REPORT_REQUEST_CODE, CrashPrompt.createReportIntent(context, crash), 0
+            )
```

Comment:
nit: We could consider moving the creation of such a pending intent behind a helper. Maybe even in `support-utils`.

Reply:
Had some issues moving this to support-util.  Couldn't find a clean way because of the Crash class dependency.

Reply:
opened #4113 to track this.